network/transport/grpc/connection_manager.go

```diff
@@ -335,53 +467,52 @@ func (s *grpcConnectionManager) openOutboundStream(connection Connection, protoc
 	if err != nil {
 		return nil, fatalError{error: fmt.Errorf("failed to read gRPC headers: %w", err)}
 	}
-	if len(peerHeaders) == 0 {
+	if len(peerHeaders) == 0 { // non-fatal error
 		return nil, fmt.Errorf("peer didn't send any headers, maybe the protocol version is not supported")
 	}
 	peerID, nodeDID, err := readMetadata(peerHeaders)
 	if err != nil {
 		return nil, fatalError{error: fmt.Errorf("failed to read peer ID header: %w", err)}
 	}
 
-	// When 2 nodes connect to each other, the connections will have to be deduplicated.
-	// When a node receives an inbound connection from a peer which it is already connected to, it must disconnect that new connection.
-	// This means the connecting side (outbound) should stop connecting to that address, because it's already connected.
-	// But it can't just clean up the outbound connection,
-	// since it's a discovered Nuts Node address which is not known by the existing inbound connection.
-	// To avoid "losing" that address it should start the outbound connector on the inbound connection,
-	// so that it can make an outbound connection when the inbound connection is closed.
-	existingConnection := s.connections.Get(ByPeerID(peerID))
-	if existingConnection != nil && existingConnection != connection {
-		connection.stopConnecting()
-		s.connections.remove(connection)
-		s.startTracking(connection.Peer().Address, existingConnection)
-		return nil, fatalError{error: ErrAlreadyConnected}
-	}
-
+	// Update connection information
 	if !connection.verifyOrSetPeerID(peerID) {
 		return nil, fatalError{error: fmt.Errorf("peer sent invalid ID (id=%s)", peerID)}
 	}
-
-	// When bootstrap node, this instance has the AcceptUnauthenticated param
-	peer := connection.Peer()
 	peerFromCtx, _ := grpcPeer.FromContext(clientStream.Context())
+	expectedPeer := connection.Peer()
 
-	authenticatedPeer := s.authenticate(nodeDID, peer, peerFromCtx)
-	connection.setPeer(authenticatedPeer)
+	// Authenticate expected DID
+	if !expectedPeer.NodeDID.Empty() { // do not authenticate bootstrap connections
+		if nodeDID.Empty() {
+			// Peer might be in maintenance mode, try again later
+			return nil, fatalError{ErrNodeDIDAuthFailed}
+		}
+		if !expectedPeer.NodeDID.Equals(nodeDID) {
+			// DID no longer lives at this address, don't call this DID again!
+			return nil, fatalError{ErrUnexpectedNodeDID} // TODO: should this also wrap ErrNodeDIDAuthFailed?
+		}
+		// Call answered by the DID we are looking for. Try to authenticate.
+		authenticatedPeer, err := s.authenticate(nodeDID, expectedPeer, peerFromCtx)
+		if err != nil {
+			return nil, fatalError{err}
```

Comment:
We're still disconnecting, changing this is for another version? Or planned for v5.1?

Reply:
Flow is unchanged from 5.0->5.1: disconnect when authentication fails. 
Failed authentication was temporarily allowed (marked as unauthenticated) after removal of `disableNodeAuthentication` config option, but reverted here since it is not backwards compatible and breaks on deduplication of connections.